### PR TITLE
Initial implementation of jujulib.URL library.

### PR DESCRIPTION
**Pull request by frankban**
It includes a charm/bundle URL object representation, including functions for rendering and parsing string URLs.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/6060/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/6061/
Test PASSed.

**Comment by anthonydillon** [comment]
Great stuff. Cant wait to start using it :+1:

File: jujugui/static/gui/src/app/jujulib/urls.js [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+    'precise',
+    'quantal',
+    'raring',
+    'saucy',
+    'trusty',
+    'utopic',
+    'vivid',
+    'wily',
+    'xenial',
+    'yakkety',
```
Comment (frankban) [inline jujugui/static/gui/src/app/jujulib/urls.js]: Yeah, done!

**Comment by frankban** [comment]
Thanks for the reviews!
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**